pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -518,6 +546,51 @@ func (a *cpuAccumulator) takeFullSockets() {
 		a.take(cpusInSocket)
 	}
 }
+func (a *cpuAccumulator) takeFullUnCore() {
+	for _, uncore := range a.freeUncoreCache() {
+		cpusInUncore := a.topo.CPUDetails.CPUsInUncoreCaches(uncore)
+		if !a.needsAtLeast(cpusInUncore.Size()) {
+			continue
+		}
+		a.take(cpusInUncore)
+	}
+}
+
+// First try to take full UncoreCache, if available and need is at least the size of the UncoreCache group.
+// Second try to take the partial UncoreCache if available and the request size can fit w/in the UncoreCache.
+func (a *cpuAccumulator) takeUncoreCache() {
+	for _, uncore := range a.sortAvailableUncoreCaches() {
+		// take full UncoreCache if the CPUs needed is greater than free UncoreCache size
+		if a.numCPUsNeeded >= a.topo.NumCPUs/a.topo.NumUncoreCache {
```

Comment:
please replace this `if` with
```
if a.needsAtLeast(a.topo.NumCPUs/a.topo.NumUncoreCache)
```
in addition: `a.topo.NumCPUs/a.topo.NumUncoreCache` is a loop invariant, should be computed once and at the top of the function and then the value reused.
Last but not least: this is the wrong way to count the (V)CPUs per Uncore group. It assumes equal uncore groups, but we know already this is overly simplified (offlined cpus is a easy example, and yes, we got actual reports about this) but still it's sadly the best we can do in the current architecture.